Nos/Service/FeatureFlags.swift

```diff
@@ -1,17 +1,25 @@
 import Foundation
 import Dependencies
 
-/// The set of feature flags used by the app.
-protocol FeatureFlags {
+/// Enum to represent each feature flag. Add new feature flags here easily.
+enum FeatureFlag: String {
     /// Whether the new media display should be enabled or not.
     /// - Note: See [#1177](https://github.com/planetary-social/nos/issues/1177) for details on the new media display.
-    var newMediaDisplayEnabled: Bool { get }
+    case newMediaDisplay
+    /// Whether the new moderation flow should be enabled or not.
+    /// - Note: See [#1489](https://github.com/planetary-social/nos/issues/1489) for details on the new moderation flow.
+    case newModerationFlow
+}
 
-    // MARK: - Additional requirements for debug mode
+/// The set of feature flags used by the app.
+protocol FeatureFlags {
+    /// Retrieves the value of the specified feature flag.
+    func isEnabled(_ feature: FeatureFlag) -> Bool
 
+    // MARK: - Additional requirements for debug mode
     #if DEBUG || STAGING
-    /// Sets the value of `newMediaDisplayEnabled`.
-    func setNewMediaDisplayEnabled(_ enabled: Bool)
+    /// Sets the value of the specified feature flag.
+    func setEnabled(_ feature: FeatureFlag, enabled: Bool)
```

Comment:
The API design of this could be improved by changing `setEnabled` to `setFeature` here:
```suggestion
    func setFeature(_ feature: FeatureFlag, enabled: Bool)
```

This way, at call sites, the word "feature" in the function name immediately precedes a feature flag enum case and the word "enabled" doesn't appear twice.